go/rep_replication.go

```diff
@@ -0,0 +1,123 @@
+package pb
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/ipfs/go-cid"
+)
+
+const header = "rtrade-replication v0"
+
+// New creates a new Replication file, with the correct header
+func New() *Replication {
+	return &Replication{
+		Header: header,
+	}
+}
+
+// Valid checks if this is an Valid Replication file, with:
+// - correct header (if strict, then no unknown extensions),
+// - at least one CID,
+// - if strict, then equal or more servers than required for replication, and
+// - replication factor greater than 1.
+// reports the first error found.
+//
+// If strict, then invalid CIDs, Servers, and unknown protocol extensions are also not allowed.
+func (r *Replication) Valid(strict bool) error {
+	if _, err := r.GetHeaderString(strict); err != nil {
+		return err
+	}
+
+	if strict && haveUnrecognized(r) {
+		return errors.New("found unrecognized data")
+	}
+
+	CIDs, err := r.GetCIDs()
+	if (strict || len(CIDs) == 0) && err != nil {
+		return err
```

Comment:
since none of these values are kept beyond this conditional check, can be simplified and made more clear like
```Go
if CIDs, err := r.GetCIDS(); err != nil && (strict || len(CIDS) == 0) {
  return err
}
```

Reply:
ok